### PR TITLE
Feature/data min freq

### PR DESCRIPTION
**Pull request by rafaelsul123**
implemented minimal frequency cutoff for spectrogram - takes place within the preprocessing, as part of _preprocessors.yaml config (after spectrogram, before amplitude_2_db)
the argument is called min_freq_spectrogram and is input through data config.
updated all data training configs with min_freq_spectrogram=0 - change as required.

### CODE REVIEW

File: soundbay/data.py [hunk 1]
```diff
@@ -288,6 +290,25 @@ def __call__(self, sample):
...
 
 
+class MinFreqSpectrogram:
+    """Cut the spectrogram frequency axis to make it start from min_freq"""
+    def __init__(self, min_freq_spectrogram, sample_rate):
+        self.min_freq_spectrogram = min_freq_spectrogram
+        self.sample_rate = sample_rate
+
+    def edit_spectrogram_axis(self, sample):
+        max_freq_in_spectrogram = self.sample_rate / 2
```
Comment (yogevmatalon) [inline soundbay/data.py:300]: maybe add a comment for using the max_freq parameter if it will be implemented in the future

**Review by yogevmatalon — CHANGES_REQUESTED** [review]
As discussed, need to check how this change affects the inference. instances with lower freq will be removed, tagged as 0, or regular prediction by the model?

**Comment by noamzbr** [comment]
> As discussed, need to check how this change affects the inference. instances with lower freq will be removed, tagged as 0, or regular prediction by the model?

I understand that this needs to be present also in the inference pipeline, but I'm not sure I followed your comment about how it will affect instances. I assumed that this stage is idependant of tagging, and that it's the user's (our) responsaiblity to make sure we're not cutting our our calls.

File: soundbay/conf/preprocessors/_preprocessors.yaml [hunk 2]
```diff
@@ -4,6 +4,10 @@ _preprocessors:
     _target_: torchaudio.transforms.Spectrogram
     n_fft: ${data.n_fft}
     hop_length: ${data.hop_length}
+  min_freq:
```
Comment (Z30G0D) [inline soundbay/conf/preprocessors/_preprocessors.yaml:7]: I think we should make it more informative.
`min_freq_filtering`?

File: soundbay/data.py [hunk 3]
```diff
@@ -288,6 +290,25 @@ def __call__(self, sample):
         return (sample - sample.min()) / (sample.max() - sample.min())
 
 
+class MinFreqSpectrogram:
+    """Cut the spectrogram frequency axis to make it start from min_freq"""
+    def __init__(self, min_freq_spectrogram, sample_rate):
```
Comment (Z30G0D) [inline soundbay/data.py:295]: need to add type check in function definition.
`sample_rate` is integer , `min_freq_spectrogram` is `Tensor`(I Think?)

**Review by Z30G0D — CHANGES_REQUESTED** [review]
Very handy.
Great job!
+1 for @yogevmatalon 's and Noam's comment.

File: soundbay/conf/preprocessors/_preprocessors.yaml (hunk 2 above)
Comment (rafaelsul123) [reply soundbay/conf/preprocessors/_preprocessors.yaml:7]: sounds right, changed the name

File: soundbay/data.py (hunk 1 above)
Comment (rafaelsul123) [reply soundbay/data.py:300]: as of current implementation, inference is not affected in any way. I can implement the filtering for inference as well, but I think that deserves a seperate implementation that can be controlled through the relevant conf files (in parallel to the training conf files and implementation)

given the current implementation (and assuming no major changes), the inference will act just like normal, using spectrograms from freq=0 to sample_rate/2, and the output will depend solely on the trained model's ability to detect the signal, wherever it is within the spectrum

*about the comment, i've added a note within data.py, inside class implementation, and also added checks to make sure that  0 < min_freq_filtering < sample_rate/2
if/when we implement a max_freq filter similar to the min_freq one, id suggest we implement a check that takes place in both of the proccesses to make sure there are no contradictions

**Comment by rafaelsul123** [comment]
hey guys, would love a second check, I tried to answer all the requested changes and would love to PR the feature :)

**Review by noamzbr — APPROVED** [review]
Approved but would like to know/add two things:
1. Just making sure - did you visually inspect the result to see that the `:-min_value` thing is the right way to go? I have no inturition why the spectorgram should start with 0 being the max freq...
2. Please also add `min_freq_filtering` to `_preprocesors_unit`

**Comment by rafaelsul123** [comment]
> Approved but would like to know/add two things:
> 
> 1. Just making sure - did you visually inspect the result to see that the `:-min_value` thing is the right way to go? I have no intuition why the spectorgram should start with 0 being the max freq...
> 2. Please also add `min_freq_filtering` to `_preprocesors_unit`

1. Checked and works as intended (saved jpg's to make sure). I agree it doesnt make sense considering the math behind the spectrogram, but is does work as intended considering the Tensor data structure (the 0,0 point is top left corner, same as np arrays)
2. Added. btw, the n_fft and hop_length there are hard coded, is this intended?

**Review by Z30G0D — APPROVED** [review]

**Review by yogevmatalon — APPROVED** [review]

**Pull request merged**